PlayerPanel.js

```diff
@@ -34,87 +34,121 @@ function update_pclist() {
 
 
 	});
-
-
-
+	const addPartyButtonContainer = $("<div class='add-party-container'></div>");
+	const addPartyButton = $("<button id='add-party'>ADD PARTY</button>");
+	addPartyButton.on('click', () => {
+		window.pcs.forEach(function (p, i) {
+			token_button({ target: $(`[data-name='${p.name}']`) }, i, window.pcs.length);
+		});
+	});
+	addPartyButtonContainer.append(addPartyButton);
+	pcs_list.append(addPartyButtonContainer);
 
 	var NEXT_COLOR = 0;
 	window.pcs.forEach(function(item, index) {
 
 		color = "#" + TOKEN_COLORS[NEXT_COLOR++];
 
 		pc = item;
-		row = $("<div/>");
-		row.css("padding", "20px");
-
-		player_name = $("<div/>");
-		player_name.text(pc.name).css("background", color).css("font-weight", "bold");
-		player_name.attr('data-nextcolor', NEXT_COLOR);
-		row.append(player_name);
-
-		row_inside = $("<div/>");
-		row_inside.css('border', '1px solid black');
-		row_inside.css("width", "100%");
-		row_inside.css("float", "left");
-		row.append(row_inside);
-
-		row_inside.append($("<div style='float:left;'><img width=60 height=60 src='" + pc.image + "'></div>"));
-
-		stats = $("<div/>");
-		stats.css("float", "left");
-
-		stats.append("<div><div><b>HP:</b></div><div class='hp'>-</div></div>");
-		stats.append("<div><div><b>AC:</b></div><div class='ac'>-</div></div>");
-		stats.append("<div><div><b>Passive Perception:</b></div><div class='pp'>-</div></div>");
-
-		row_inside.append(stats);
-
-		conditions = $("<div style='float:left;'/>");
-		conditions.append("<div><b>Conditions</b></div>");
-		conditions.append("<div class='conditions'></div>");
-
-		row_inside.append(conditions);
 
+		let playerData;
 		if (pc.sheet in window.PLAYER_STATS) {
-			data = window.PLAYER_STATS[pc.sheet];
-
-			stats.find(".hp").text(data.hp + "/" + data.max_hp);
-			stats.find(".ac").text(data.ac);
-			stats.find(".pp").text(data.pp);
-
-			for (var i = 0; i < data.conditions.length; i++)
-				conditions.find(".conditions").append($("<div/>").text(data.conditions[i]));
+			playerData = window.PLAYER_STATS[pc.sheet];
 		}
+		
+		newPlayerTemplate = `
+			<div class="player-card">

```

Comment:
Just in general here, a lot of these string literal expressions, you'd want to assign to a temp variable, above building the string, then place the just the var name into the HTML.

This will clean things up a bit and separate the data/UI layers.

Another advantage would be that using string literals like this actually saves the whitespace, which isn't something we necessarily want en client. I forget if this keeps the indentation without context that we're already multiple columns into the file.

Reply:
It would be great to build the UI with Vue or React, for this template management (my choice would be Vue).
By doing so we will keep the presentation and the logic separate, following the MVC principles.

But I am not the maintainer and I understand this would add another layer of complexity to a so far simple tool. 
It might require a better discussion than comments on a PR, imho.